crates/router/src/connector/paybox/transformers.rs

```diff
@@ -102,121 +435,345 @@ impl From<PayboxPaymentStatus> for enums::AttemptStatus {
     }
 }
 
-//TODO: Fill the struct with respective fields
 #[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
-pub struct PayboxPaymentsResponse {
-    status: PayboxPaymentStatus,
-    id: String,
+pub struct PayboxResponse {
+    #[serde(rename = "NUMTRANS")]
+    pub transaction_number: String,
+
+    #[serde(rename = "NUMAPPEL")]
+    pub call_number: String,
+
+    #[serde(rename = "CODEREPONSE")]
+    pub response_code: String,
+
+    #[serde(rename = "COMMENTAIRE")]
+    pub response_message: String,
+}
+
+pub fn parse_url_encoded_to_struct(
+    query_bytes: Bytes,
+) -> Result<PayboxResponse, errors::ConnectorError> {
+    let query_string = String::from_utf8_lossy(&query_bytes);
+    let parsed: std::collections::HashMap<String, String> =
+        url::form_urlencoded::parse(query_string.as_bytes())
+            .map(|(k, v)| (k.to_string(), v.to_string()))
+            .collect();
+
+    let json_string =
+        serde_json::to_string(&parsed).map_err(|_e| errors::ConnectorError::ParsingFailed)?;
+
+    let response: PayboxResponse =
+        serde_json::from_str(&json_string).map_err(|_e| errors::ConnectorError::ParsingFailed)?;
+
+    Ok(response)
+}
```

Comment:
Get this reviewed from @SanchithHegde

Reply:
These parsing failures will trigger internal server error at our application

Reply:
that is expected and we are doing same for all other connectors if it failed to deserialize the response we are throwing the error ParsingFailed